dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoaderRef.java

```diff
@@ -0,0 +1,160 @@
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.classloading.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.security.ProtectionDomain;
+import java.util.EnumSet;
+import java.util.Enumeration;
+import java.util.Objects;
+
+import org.osgi.framework.Bundle;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.wsspi.classloading.ApiType;
+
+/**
+ * This class is used to create either weak references to AppClassLoaders
+ */
+@Trivial
+public class ClassLoaderRef extends LibertyLoader implements SpringLoader {
+    static {
+        ClassLoader.registerAsParallelCapable();
+    }
+
+    private final WeakReference<AppClassLoader> classLoaderWeakRef;
+
+    ClassLoaderRef(AppClassLoader appClassLoader) {
+        super(appClassLoader.getParent());
+        classLoaderWeakRef = new WeakReference<>(appClassLoader);
+    }
+
+    @Override
+    protected final Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        return loadClass(name, resolve, false, false);
+    }
+
+    @Override
+    public URL getResource(String resName) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResource(resName) : parent.getResource(resName);
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String resName) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResourceAsStream(resName) : parent.getResourceAsStream(resName);
+    }
+
+    @Override
+    public Enumeration<URL> getResources(String resName) throws IOException {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResources(resName) : parent.getResources(resName);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof ClassLoaderRef) {
+            return Objects.equals(classLoaderWeakRef.get(), ((ClassLoaderRef) o).classLoaderWeakRef.get());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean addTransformer(ClassFileTransformer cft) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        if (appCL != null) {
+            return appCL.addTransformer(cft);
+        }
+        return false;
+    }
+
+    @Override
+    public ClassLoader getThrowawayClassLoader() {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getThrowawayClassLoader() : parent;
```

Comment:
Would returning `this` be more appropriate when `appCL` is `null`?

Reply:
Will do